src/main/kotlin/com/cognifide/gradle/aem/pkg/ComposeTask.kt

```diff
@@ -245,18 +253,32 @@ open class ComposeTask : Zip(), AemTask {
         }
     }
 
-    fun dependProject(projectPath: String, taskNames: Set<String>) {
+    fun dependProject(projectPath: String, taskNames: Collection<String>) {
         dependProject(project.findProject(projectPath), taskNames)
     }
 
-    fun dependProject(project: Project, taskNames: Set<String>) {
-        taskNames.forEach { taskName -> dependsOn("${project.path}:$taskName") }
+    fun dependProject(project: Project, taskNames: Collection<String>) {
```

Comment:
closures in groovy are not serializable so that I decided to introduce mechanism for suffix added to task name '.dependencies' will effectively cause that it will be replaced by concrete tasks names of dependant tasks